File: Jukebox.java
/*
Now that we have that out of the way, we'll outline the basic system components.
• Jukebox
• CD
• Song
• Artist
• Playlist
• Display (displays details on the screen)
Now, let’s break this down further and think about the possible actions.
• Playlist creation (includes add, delete, and shuffle)
• CD selector
• Song selector
• Queuing up a song
• Get next song from playlist
A user also can be introduced:
• Adding
• Deleting
• Credit information
*/

public class CD {
}
 
public class CDPlayer {
    private Playlist p;
    private CD c;
 
    public Playlist getPlaylist() {
        return p;
    }
 
    public void setPlaylist(Playlist p) {
        this.p = p;
    }
 
    public CD getCD() {
        return c;
    }
 
    public void setCD(CD c) {
        this.c = c;
    }
 
    public CDPlayer(Playlist p) {
        this.p = p;
    }
 
    public CDPlayer(CD c, Playlist p) {
        // TODO
    }
 
    public CDPlayer(CD c) {
        this.c = c;
    }
 
    public void playTrack(MySong s) {
        // TODO
    }
}
 
public class JukeBox {
    private CDPlayer cdPlayer;
    private User user;
    private Set<CD> cdCollection;
    private TrackSelector ts;
 
    public JukeBox(CDPlayer cdPlayer, User user, Set<CD> cdCollection,
            TrackSelector ts) { // TODO
 
    }
 
    public MySong getCurrentTrack() {
        return ts.getCurrentSong();
    }
 
    public void processOneUser(User u) {
        this.user = u;
    }
}
 
public class Playlist {
    private MySong track;
    private Queue<MySong> queue;
 
    public Playlist(MySong track, Queue<MySong> queue) { // TODO
 
    }
 
    public MySong getNextTrackToPlay() {
        return queue.peek();
    }
 
    public void queueUpTrack(MySong s) {
        queue.add(s);
    }
}
 
public class Song {
    private String songName;
}
 
public class TrackSelector {
    private MySong currentSong;
 
    public TrackSelector(MySong s) {
        currentSong = s;
    }
 
    public void setTrack(MySong s) {
        currentSong = s;
    }
 
    public MySong getCurrentSong() {
        return currentSong;
    }
}
 
public class User {
    private String name;
 
    public String getName() {
        return name;
    }
 
    public void setName(String name) {
        this.name = name;
    }
 
    public long getID() {
        return ID;
    }
 
    public void setID(long iD) {
        ID = iD;
    }
 
    private long ID;
 
    public User(String name, long iD) { // TODO
 
    }
 
    public User getUser() {
        return this;
    }
 
    public User addUser(String name, long iD) {// TODO
        return null;
    }
}

File: Jukebox.py
"""
Now that we have that out of the way, we'll outline the basic system components.
• Jukebox
• CD
• Song
• Artist
• Playlist
• Display (displays details on the screen)
Now, let’s break this down further and think about the possible actions.
• Playlist creation (includes add, delete, and shuffle)
• CD selector
• Song selector
• Queuing up a song
• Get next song from playlist
A user also can be introduced:
• Adding
• Deleting
• Credit information
"""

#Music Jukebox
"""
User,CD,Song
"""
import collections
class User:
    def __init__(self):
        self.name = None
        self.userID = None
    
    # Getters and Setters for username and userID

class Song:
    """
    songname, artist name,id,length,CD
    """

class CD:
    """
    id,artist,songs
    """
"""
The Playlist manages the current and next songs to play. It is essentially a wrapper
class for a queue and offers some additional methods for convenience.
"""
class PlayList:
    def __init__(self,song,queue):
        self.song = song
        self.queue = queue #collections.deque()
        
    def queueUpSong(self,song):
        self.queue.append(song)
    
    def peekSong(self):
        return self.queue[0]
    
"""
Like a real CD player, the CDPlayer class supports storing just one CD at a time. The
CDs that are not in play are stored in the jukebox.
"""

class CDPlayer:
    def __init__(self,playList,cd):
        self.playList = playList
        self.cd = cd
    def playSong(self,song):
        #logic for play song
    """
    Getters and Setters for playList and cd {getPlaylist,getCd...}
    """
"""
UserManager
"""
class UserManager:
    def __init__(self,v = []):
        self.userList = v
    """
    Getters and Setters for userList
    """
    def addUser(self,user):
        self.userList.append(user)
    def removeUser(self,user):
        self.userList.remove(user)

"""
The Jukebox class represents the body of the problem. Many of the interactions
between the components of the system, or between the system and the user, are
channeled through here.
"""

class Jukebox:
    def __init__(self,user =None,playList = None,cds = set(),CDPlayer = None):
        self.user = user
        #...
    #Getters and Setters for parameters
    
    def getCurrentSong(self):
        #Logic

"""
Additional Payment class -> can be abstract class and payment methods such as coins or card
Inventory class for maintaining payment information
coin class etc
"""
    
